stdlib/public/core/Integers.swift.gyb

```diff
@@ -1661,22 +1661,145 @@ ${assignmentOperatorComment(x.nonMaskingOperator, False)}
+  ///
+  /// - `.toNearestOrEven` and `.toNearestOrAwayFromZero` are useful if you
+  ///   need to emulate or implement floating-point rounding with integer
+  ///   arithmetic.
+  ///
+  /// - Parameters:
+  ///   - divisor: the number to divide by.
+  ///   - rule: the rounding rule to use.
+  ///
+  /// See also `remainder(dividingBy:, roundingQuotient:)` and
+  /// `quotientAndRemainder(dividingBy:, roundingQuotient:)`.
+  func divided(by divisor: Self, rounding rule: RoundingRule) -> Self
+
+  /// The remainder of dividing this value by `divisor`, using the specified
+  /// rounding `rule`.
+  ///
+  /// Equivalent to
+  /// ~~~~
+  /// let quotient = self.divided(by: divisor, rounding: rule)
+  /// return self - q*quotient
+  /// ~~~~
+  ///
+  /// A few of the available rounding rules are worth calling out in particular:
+  ///
+  /// - `.towardZero` matches the behavior of the `%` operator.
+  ///
+  /// - `.down` has the property the sign of the result always matches the
+  ///   sign of the divisor, and so when the divisor is positive, the result
+  ///   is always in `0 ..< divisor`.
+  ///
+  /// - `.toNearestOrEven` and `.toNearestOrAwayFromZero` are sometimes used
+  ///   because they produce the smallest magnitude remainder.
+  ///
+  /// - Parameters:
+  ///   - divisor: the number to divide by to get the remainder.
+  ///   - rule: the rounding rule to apply to the quotient.
+  ///
+  /// See also `mod(_:)`, `divided(by:,rounding:)`,
+  /// `quotientAndRemainder(dividingBy:, roundingQuotient:)`, and `isDivisible(by:)`.
+  func remainder(
+    dividingBy divisor: Self,
+    roundingQuotient rule: RoundingRule
+  ) -> Self
+
+  /// The mathematical modulo operation.
+  ///
+  /// This operation is similar to the `%` operator or `.remainder`, but
+  /// differs importantly in how it handles negative numbers. The `modulus`
+  /// must be positive, and the result is always non-negative, even if `self`
+  /// is negative.
+  ///
+  /// - Parameters:
+  ///   - modulus: must be positive.
+  /// - Returns: the unique integer in `0 ..< modulus` that satisfies
+  ///   `self = n*modulus + result` for some integer `n`.
+  ///
+  /// See also `remainder(dividingBy:, roundingQuotient:)` and `isDivisible(by:)`.
+  func mod(_ modulus: Self) -> Self
+
+  /// Shifts this value right by n, rounding according to the specified
+  /// rounding rule.
+  ///
+  /// - Parameters:
+  ///   - n: The number of bits to shift by. Must be non-negative.
+  ///   - rule: The direction in which to round the result if it is not exact.
+  /// - Returns: the pair `(rounded: Self, exact: Bool)`, where `rounded`
+  ///   is `self*2**(-n)` rounded to the nearest integer, and `exact` is
+  ///   true if and only if the result of the shift was an integer before
+  ///   rounding.
+  func shiftedRight<Count>(by n: Count, rounding rule: RoundingRule)
+    -> (result: Self, isExact: Bool) where Count : BinaryInteger
```

Comment:
Do you have any examples of this in use? From a naming perspective, it might be good to put "bit" in the signature. I think `bitShiftedRight(by:)` would be good, but at minimum we could change `n` to `bits` or `bitCount`.

Reply:
I have some other branches in preparation that need it, I'll try to dust off a good example from those.

Reply:
Thanks, missed that!